elephant/phase_analysis.py

```diff
@@ -82,107 +92,77 @@ def spike_triggered_phase(hilbert_transform, spiketrains, interpolate):
-
-        # Take only spikes which lie directly within the signal segment -
-        # ignore spikes sitting on the last sample
-        sttimeind = np.where(np.logical_and(
-            spiketrain >= start[phase_i], spiketrain < stop[phase_i]))[0]
-
-        # Find index into signal for each spike
-        ind_at_spike = np.round(
-            (spiketrain[sttimeind] - hilbert_transform[phase_i].t_start) /
-            hilbert_transform[phase_i].sampling_period). \
-            simplified.magnitude.astype(int)
-
-        # Extract times for speed reasons
-        times = hilbert_transform[phase_i].times
-
-        # Append new list to the results for this spiketrain
-        result_phases.append([])
-        result_amps.append([])
-        result_times.append([])
-
-        # Step through all spikes
-        for spike_i, ind_at_spike_j in enumerate(ind_at_spike):
-            # Difference vector between actual spike time and sample point,
-            # positive if spike time is later than sample point
-            dv = spiketrain[sttimeind[spike_i]] - times[ind_at_spike_j]
-
-            # Make sure ind_at_spike is to the left of the spike time
-            if dv < 0 and ind_at_spike_j > 0:
-                ind_at_spike_j = ind_at_spike_j - 1
+    for spiketrain_i in range(num_spiketrains):
+        full_sp = spiketrains[spiketrain_i]
+        for trial_i in range(num_trials):
+            spike_times = full_sp.time_slice(t_starts[trial_i], t_stops[trial_i]).times  # leave out last spike
+            time = times[trial_i]
+            phase = phases[trial_i]
+            amp = amps[trial_i]
+
+            # cut righthand border
+            if spike_times[-1] == time[-1]:
+                spike_times = spike_times[:-1]
 
             if interpolate:
-                # Get relative spike occurrence between the two closest signal
-                # sample points
-                # if z->0 spike is more to the left sample
-                # if z->1 more to the right sample
-                z = (spiketrain[sttimeind[spike_i]] - times[ind_at_spike_j]) /\
-                    hilbert_transform[phase_i].sampling_period
-
-                # Save hilbert_transform (interpolate on circle)
-                p1 = np.angle(hilbert_transform[phase_i][ind_at_spike_j])
-                p2 = np.angle(hilbert_transform[phase_i][ind_at_spike_j + 1])
-                result_phases[spiketrain_i].append(
-                    np.angle(
-                        (1 - z) * np.exp(np.complex(0, p1)) +
-                        z * np.exp(np.complex(0, p2))))
-
-                # Save amplitude
-                result_amps[spiketrain_i].append(
-                    (1 - z) * np.abs(
-                        hilbert_transform[phase_i][ind_at_spike_j]) +
-                    z * np.abs(hilbert_transform[phase_i][ind_at_spike_j + 1]))
+                # use the scipy.interp1d fits to get phases at spiketimes
+                phase_fit = interp1d(time, phase, kind='linear')
```

Comment:
I have not verified if this is actually valid. What happens when the phase goes from ~-pi to ~+pi?